DQMOffline/L1Trigger/interface/L1TStage2CaloLayer2Offline.h

```diff
@@ -0,0 +1,183 @@
+#include "DataFormats/L1Trigger/interface/EtSum.h"
+
+class L1TStage2CaloLayer2Offline: public DQMEDAnalyzer {
+
+public:
+
+  L1TStage2CaloLayer2Offline(const edm::ParameterSet& ps);
+  virtual ~L1TStage2CaloLayer2Offline();
+
+protected:
+
+  void dqmBeginRun(edm::Run const &, edm::EventSetup const &) override;
+  void bookHistograms(DQMStore::IBooker &, edm::Run const &, edm::EventSetup const &) override;
+  void analyze(edm::Event const& e, edm::EventSetup const& eSetup);
+  void beginLuminosityBlock(edm::LuminosityBlock const& lumi, edm::EventSetup const& eSetup);
+  void endLuminosityBlock(edm::LuminosityBlock const& lumi, edm::EventSetup const& eSetup);
+  void endRun(edm::Run const& run, edm::EventSetup const& eSetup);
+
+private:
+  //histos booking function
+  void bookHistos(DQMStore::IBooker &);
+  void bookEnergySumHistos(DQMStore::IBooker &);
+  void bookJetHistos(DQMStore::IBooker &);
+
+  void fillEnergySums(edm::Event const& e, const unsigned int nVertex);
+  void fillJets(edm::Event const& e, const unsigned int nVertex);
+
+  //private variables
+  math::XYZPoint PVPoint_;
+
+  //variables from config file
+  edm::EDGetTokenT<reco::CaloJetCollection> theCaloJetCollection_;
+  edm::EDGetTokenT<reco::CaloMETCollection> thecaloMETCollection_;
+  edm::EDGetTokenT<reco::VertexCollection> thePVCollection_;
+  edm::EDGetTokenT<reco::BeamSpot> theBSCollection_;
+  edm::EDGetTokenT<trigger::TriggerEvent> triggerEvent_;
+  edm::EDGetTokenT<edm::TriggerResults> triggerResults_;
+  edm::InputTag triggerFilter_;
+  std::string triggerPath_;
+  std::string histFolder_;
+  std::string efficiencyFolder_;
+
+  edm::EDGetTokenT<l1t::JetBxCollection> stage2CaloLayer2JetToken_;
+  edm::EDGetTokenT<l1t::EtSumBxCollection> stage2CaloLayer2EtSumToken_;
+
+  std::vector<double> jetEfficiencyThresholds_;
+  std::vector<double> metEfficiencyThresholds_;
+  std::vector<double> mhtEfficiencyThresholds_;
+  std::vector<double> ettEfficiencyThresholds_;
+  std::vector<double> httEfficiencyThresholds_;
+
+  std::vector<double> jetEfficiencyBins_;
+  std::vector<double> metEfficiencyBins_;
+  std::vector<double> mhtEfficiencyBins_;
+  std::vector<double> ettEfficiencyBins_;
+  std::vector<double> httEfficiencyBins_;
+
+  // TODO: add turn-on cuts (vectors of doubles)
+  // Histograms
+  MonitorElement* h_nVertex_;
+
+  // energy sums reco vs L1
+  MonitorElement* h_L1METvsCaloMET_;
+  MonitorElement* h_L1MHTvsRecoMHT_;
+  MonitorElement* h_L1METTvsCaloETT_;
+  MonitorElement* h_L1HTTvsRecoHTT_;
+
+  MonitorElement* h_L1METPhivsCaloMETPhi_;
+  MonitorElement* h_L1MHTPhivsRecoMHTPhi_;
+
+  // energy sum resolutions
+  MonitorElement* h_resolutionMET_;
+  MonitorElement* h_resolutionMHT_;
+  MonitorElement* h_resolutionETT_;
+  MonitorElement* h_resolutionHTT_;
+  MonitorElement* h_resolutionMETPhi_;
+  MonitorElement* h_resolutionMHTPhi_;
+
+  // energy sum turn ons
+  std::map<double, MonitorElement*> h_efficiencyMET_pass_;
```

Comment:
The idea of the `<double, MonitorElement>` map is to simplify the creation of efficiency histograms for different *threshold*. E.g. You want to plot the MET efficiency curves for different L1 triggers, say 50, 100, 150 and 200 GeV. Given the vectors of thresholds (read from python config) one can do this in a simple for-loop. An example is included in this PR, in the bit that is not automatically loaded on github: [L1TStage2CaloLayer2Offline.cc#L214](https://github.com/kreczko/cmssw/blob/877b21c84eb0e4e3fea28fc0d49e03d73afb99cf/DQMOffline/L1Trigger/src/L1TStage2CaloLayer2Offline.cc#L214)